src/lib/api.js

```diff
@@ -76,6 +76,27 @@ const extractBbox = function (params) {
   return undefined
 }
 
+const extractLimit = function (params) {
+  const { limit: limitStr } = params
+
+  if (limitStr !== undefined) {
+    let limit
+    try {
+      limit = parseInt(limitStr)
+    } catch (e) {
+      throw new ValidationError('Invalid limit value')
+    }
+
+    if (Number.isNaN(limit) || limit <= 0 || limit > 10000) {
```

Comment:
What is the realistic page limit here?   Before it couldn't be much above 500, but now with compression turned on it should be a lot better.

Reply:
I'm going to need to think on that some. filed https://github.com/stac-utils/stac-server/issues/226 so we don't forget.